db/queries/base.py

```diff
@@ -1,203 +1,145 @@
-from collections import namedtuple
-
-from sqlalchemy import select, join, Column as SAColumn, Table as SATable
-
-from django.utils.functional import cached_property
+from sqlalchemy import MetaData, select
 
 from db.records.operations import select as records_select
 from db.columns.base import MathesarColumn
+from db.columns.operations.select import get_column_name_from_attnum
+from db.tables.operations.select import reflect_table_from_oid
 from db.transforms.operations.apply import apply_transformations
 
 
 class DBQuery:
     def __init__(
-        self,
-        base_table,
-        initial_columns,
-        transformations=None,
-        name=None
+            self,
+            base_table_oid,
+            initial_columns,
+            engine,
+            transformations=None,
+            name=None,
     ):
-        assert isinstance(base_table, SATable)
-        self.base_table = base_table
+        self.base_table_oid = base_table_oid
         for initial_col in initial_columns:
             assert isinstance(initial_col, InitialColumn)
         self.initial_columns = initial_columns
+        self.engine = engine
```

Comment:
I somehow feel uneasy about engine being an attribute of `DBQuery`. Can't articulate further. Did you make this change just for convenience, or was it some overarching principle?

Reply:
I realized that the changes resulted in every public method of the class requiring an engine, and since we have a global engine anyway, it seemed reasonable to just instantiate with it. I didn't see a use case for passing different engines to different methods of the same query, since they should all use the same connection string for a given "conceptual" query, anyway.